doc/source/api.rst

```diff
@@ -0,0 +1,179 @@
+*****************
+API Documentation
+*****************
+
+We use the :class:`Broker` to pose queries for saved data sets ("runs"). A
+search query returns a lazy-loaded iterable of :class:`Header` objects.
+Each :class:`Header` encapsulates the metadata for one run. It provides
+convenient methods for exploring that metadata and loading the full data.
+
+.. currentmodule:: databroker
+
+The Broker object
+-----------------
+
+Making a Broker
+===============
+
+You can instantiate a :class:`Broker` by passing it a dictionary of
+configuration or by providing the name of a configuration file on disk.
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Broker.from_config
```

Comment:
I think this is a much convenient interface for configuration setup at different beamlines.